pkgs/build-support/libredirect/default.nix

```diff
@@ -32,6 +32,14 @@ else stdenv.mkDerivation rec {
   buildPhase = ''
     runHook preBuild
 
+    ENABLE_TEST_SYSTEM=1
+    if [ "$(cat /proc/self/uid_map)" = "         0          0 4294967295" ]; then
+      echo "Boot namespace detected! Disabling tests that use system()."
+      echo "The system() relies on /bin/sh which may not be compatible with nixpkgs's"
+      echo "glibc, so we can't test this without the sandboxing isolation."
+      ENABLE_TEST_SYSTEM=0
+    fi
```

Comment:
Also, could this be moved to the `preCheck` block somehow?  That would be easier to read.  It is testing the wrong thing on `host!=build` cross compiles, but we disable the `checkPhase` there anyways so I guess it doesn't matter.

Reply:
The checkPhase isn't run for cross builds. Moving this check to the `checkPhase` is not a bad idea.